test/unit/math/test_ad.hpp

```diff
@@ -1493,24 +1596,11 @@ void expect_ad_matvar(F&& f, const T& x) {
       SUCCEED();
     }
   }
-  for (Eigen::Index i = 0; i < A_vm.size(); ++i) {
-    A_vm_f.adj()(i) = 1;
-    A_mv_f.adj()(i) = 1;
-    stan::math::grad();
-    EXPECT_FLOAT_EQ(A_vm.adj()(i), A_mv.adj()(i));
-    EXPECT_FLOAT_EQ(A_vm_f.adj()(i), A_mv_f.adj()(i));
-    stan::math::set_zero_all_adjoints();
-  }
-  var b_mv = sum(A_mv_f);
-  var b_vm = sum(A_vm_f);
-  var final_var = b_mv + b_vm;
-  stan::math::grad(final_var.vi_);
-  if (!is_finite(x) || !stan::math::isfinite(final_var)) {
+  test_matvar_gradient(A_mv, A_vm, A_mv_f, A_vm_f);
+  if (!x.allFinite()) {
```

Comment:
Why here? I'm surprised you'd need this. If `test_matvar_gradient` passes I'm surprised `test_matvar_sum_gradient` wouldn't.

Reply:
Oh actually this should be above `test_matvar_gradient()` like in `test_gradient()`